gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/spring/security/SecurityConfiguration.java

```diff
@@ -228,6 +235,33 @@ private HttpSecurity csrf(HttpSecurity security) throws Exception {
         }
     }
 
+    private void csp(HttpSecurity security) throws Exception {
+        if(environment.getProperty(HTTP_CSP_ENABLED, Boolean.class, false)) {
+            final List<String> directives = getDirectives();
+            if (directives.isEmpty()) {
+                directives.add(DEFAULT_CSP_DIRECTIVE);
+            }
+            security.headers()
+                    .contentSecurityPolicy(directives.stream()
+                            .map(directive -> directive.trim().endsWith(";") ? directive : directive + ";")
+                            .collect(Collectors.joining(" ")));
+        }
+    }
+
+    private List<String> getDirectives() {
```

Comment:
I am approving this pull request with this suggestion: better if we could avoid `break`. This below code should be equivalent 


private List<String> getDirectives() {

    final List<String> directives = new ArrayList<>();
    var value = "";
    var i = 0;
    do {
        var propertyKey = String.format(HTTP_CSP_DIRECTIVES, i);
        value = environment.getProperty(propertyKey, String.class);
        if (nonNull(value)) {
            directives.add(value);
            i++;
        }
    } while (nonNull(value));

    return directives;
}